### ISSUE TITLE
Popper mistakenly marked as reference hidden

### ISSUE THREAD

**Issue by panlina** [floating-ui/floating-ui#1940]
I'm sorry I cannot put a codesandbox here. I'm developing a chrome extension so the page in question is from 3rd party. The page has a rather deep DOM tree and I've tried to mimic a similar structure in a new page, but failed to repro the issue.

So to show this issue, I created a script that you can paste into the devtool on the website.

The page is at https://www.ey.com/en_gl and the script is below. The version I'm using is 2.11.6.

```js
var Popper = await import('https://esm.sh/@popperjs/core@2.11.6');

// click the 'Services' button on nav bar
document.querySelector('.main-navigation__l1-list > li:nth-child(3) > button').click();

// wait for the dropdown menu to come out and create popper
setTimeout(() => {
	var reference = document.querySelector('.main-navigation__l2s2-item:first-child > button');
	var popper = createPopperElement();
	Popper.createPopper(reference, popper);
}, 1000);

function createPopperElement() {
	var div = document.createElement('div');
	div.style.backgroundColor = 'white';
	div.style.zIndex = 12345;
	div.innerText = "abc";
	document.body.appendChild(div);
	return div;
}
```

Basically it clicks the 'Services' button on nav bar, waits for the dropdown menu to come out and creates a popper beside the first menu item.

The problem is that, the popper is created successfully, but it is marked `data-popper-reference-hidden` while the menu item is obviously visible, which makes my extension fail to point out the menu item.

<img width="691" alt="image" src="https://user-images.githubusercontent.com/18240847/199296268-b389108a-d1ea-4d8b-a541-d1bd289db9a7.png">

Is it some problem with popper?

**Comment by atomiks** [comment]
We have now released [Floating UI v1](https://floating-ui.com/docs/motivation), the successor to Popper.

Don't worry, Popper 2 will continue to receive new releases with bug/security fixes provided by external contributors, but development of new features is not continuing at this point. We are also closing bugs related to it as us maintainers are focusing our energy on tracking issues with Floating UI, however you are welcome to create new pull requests to fix bugs in Popper, which we will review and merge, as many packages (such as MUI, Bootstrap, etc) still rely on it. 

Thanks!
